gridstatus/spp.py

```diff
@@ -272,6 +297,311 @@ def get_interconnection_queue(self, verbose=False):
+        df = df[
+            [
+                "Time",
+                "Market",
+                "Location",
+                "Location Type",
+                "LMP",
+                "Energy",
+                "Congestion",
+                "Loss",
+            ]
+        ]
+        df = df.reset_index(drop=True)
+        return df
+
+    @staticmethod
+    def _parse_gmt_interval_end(df, interval_duration: pd.Timedelta, timezone):
+        return df["GMTIntervalEnd"].apply(
+            lambda x: (
+                pd.Timestamp(x, unit="ms", tz="UTC") - interval_duration
+            ).tz_convert(timezone),
+        )
+
+    @staticmethod
+    def _parse_day_ahead_hour_end(df, timezone):
+        # 'DA_HOUREND': '12/26/2022 9:00:00 AM',
+        return df["DA_HOUREND"].apply(
+            lambda x: (pd.Timestamp(x, tz=timezone) - pd.Timedelta(hours=1)),
+        )
+
+    @staticmethod
+    def _normalize_location_type(location_type):
+        norm_location_type = location_type.upper()
+        if norm_location_type in (LOCATION_TYPE_HUB, LOCATION_TYPE_INTERFACE):
+            return norm_location_type
+        else:
+            raise NotSupported(f"Invalid location_type {location_type}")
+
+    @staticmethod
+    def _get_location_type_name(location_type):
+        if location_type == LOCATION_TYPE_HUB:
+            return "Hub"
+        elif location_type == LOCATION_TYPE_INTERFACE:
+            return "Interface"
+        else:
+            raise ValueError(f"Invalid location_type: {location_type}")
+
+    def _get_location_list(self, location_type, verbose=False):
+        if location_type == LOCATION_TYPE_HUB:
+            df = self._get_feature_data(QUERY_RTM5_HUBS_URL, verbose=verbose)
+        elif location_type == LOCATION_TYPE_INTERFACE:
+            df = self._get_feature_data(QUERY_RTM5_INTERFACES_URL, verbose=verbose)
+        else:
+            raise ValueError(f"Invalid location_type: {location_type}")
+        return df["SETTLEMENT_LOCATION"].unique().tolist()
+
+    def _fs_get_rtbm_lmp_by_location_paths(self, date, verbose=False):
+        """Lists files for Real-Time Balancing Market (RTBM), Locational Marginal Price (LMP) by Settlement Location (SL)"""
+        if date == "latest":
+            paths = ["/RTBM-LMP-SL-latestInterval.csv"]
+        elif utils.is_today(date, self.default_timezone):
+            files_df = self._file_browser_list(
+                name=FS_RTBM_LMP_BY_LOCATION,
+                fs_name=FS_RTBM_LMP_BY_LOCATION,
+                type="folder",
+                path=date.strftime("/%Y/%m/By_Interval/%d"),
+            )
+            paths = files_df["path"].tolist()
+        if verbose:
+            print(f"Found {len(paths)} files for {date}", file=sys.stderr)
+        return paths
+
+    def _fetch_rtbm_lmp_by_location(self, date, verbose=False):
+        all_dfs = []
+        paths = self._fs_get_rtbm_lmp_by_location_paths(date, verbose=verbose)
+        for path in paths:
```

Comment:
what do you think about adding a `tqdm` progress bar here? I do that in other places where there is a lot of files that need to de downloaded leading to longer than normal runtimes. Even though there is no way for the user to turn it off, i think it is better to have stuff printed to stdout than to wait at screen that isn't updating

```python
for path in tqdm.tqdm(paths):
    ...
```

example: https://github.com/kmax12/gridstatus/blob/d47ff339930d35c5fd73ef6aea80039eb47ad751/gridstatus/pjm.py#L406

Reply:
Will add